src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,132 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    class Node<T> {
+        private Node<T> prev;
+        private T value;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = newNode;
+        } else {
+            newNode.prev = tail;
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+            size++;
+        } else {
+            Node<T> node = getNodeByIndex(index);
+            Node<T> newNode = new Node<>(node.prev, value, node);
+            node.prev.next = newNode;
+            node.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value: list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> newNode = getNodeByIndex(index);
+        return newNode.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T valueNode = getNodeByIndex(index).value;
+        getNodeByIndex(index).value = value;
```

Comment:
are we realy need to call this method here? or better will be use valueNode.value = value?)